clippy_lints/src/attrs/duplicated_attributes.rs

```diff
@@ -1,18 +1,21 @@
 use super::DUPLICATED_ATTRIBUTES;
 use clippy_utils::diagnostics::span_lint_and_then;
+use itertools::Itertools;
 use rustc_ast::{Attribute, MetaItem};
 use rustc_data_structures::fx::FxHashMap;
 use rustc_lint::EarlyContext;
-use rustc_span::{Span, sym};
+use rustc_span::{Span, Symbol, sym};
 use std::collections::hash_map::Entry;
 
 fn emit_if_duplicated(
     cx: &EarlyContext<'_>,
     attr: &MetaItem,
     attr_paths: &mut FxHashMap<String, Span>,
-    complete_path: String,
+    parent: &[Symbol],
+    complete_path: impl FnOnce(&str) -> String,
 ) {
-    match attr_paths.entry(complete_path) {
+    let parent = parent.iter().join(":");
+    match attr_paths.entry(complete_path(&parent)) {
```

Comment:
Is there a reason for this closure-based change? The join + closure call (which invokes `format!()`) looks like it's doing essentially the same thing as before which had the `format!()` directly inlined as the function argument, or am I missing a subtle difference? (The change from `Vec<String>` to `Vec<Symbol>` is nice though)

Reply:
Ah, reading the other comment from Alexendoo, I guess the variable declaration line above used to be way longer and you wanted to share that between the two calls? It's not really much longer anymore now and the closure makes it slightly harder to tell what's going on but that's a bit pedantic, if you still think that's an improvement then I'm fine merging it as is too